Task "Callback Refactoring"

kapcta opened the pull request:
Code was re-written using Strategy pattern.
Also enum CallbackData.java was added for this operations

kapcta committed:
Task "Callback Refactoring"
Code was re-written using Strategy pattern.
Also enum CallbackData.java was added for this operations

ILevchugov commented on src/main/java/io/levchugov/petproject/handler/strategy/AddMovieStrategy.java:
```diff
@@ -0,0 +1,24 @@
+import java.io.Serializable;
+
+import static io.levchugov.petproject.repository.InMemoryStorage.AWAITS_TITLE;
+
```
```suggestion
@Component
```

ILevchugov commented on src/main/java/io/levchugov/petproject/handler/strategy/CallbackData.java:
```diff
@@ -0,0 +1,21 @@
+package io.levchugov.petproject.handler.strategy;
+
```
```suggestion
@Getter
@RequiredArgsConstructor
```

ILevchugov commented on src/main/java/io/levchugov/petproject/handler/strategy/Strategy.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.io.Serializable;
+
+public interface Strategy {
```
May be this name represents a logic of interface better? 
```suggestion
public interface CallBackProcessStrategy {
```

ILevchugov commented on src/main/java/io/levchugov/petproject/config/StrategiesConfig.java:
```diff
@@ -0,0 +1,58 @@
+@Configuration
+public class StrategiesConfig {
+
+    @Bean
```
You could use annotation Component in this classes instead of Bean

kapcta committed:
Task "Callback Refactoring"
@Component annotation was added where it needed and configuration file was changed according to changes abovementioned. Also name of Strategy interface was re-named to CallBackProcessStrategy. StrategiesConfig.java was deleted!!!

kapcta committed:
Task "Callback Refactoring"
@Component annotation was added where it needed and configuration file was changed according to changes abovementioned. Also name of Strategy interface was re-named to CallBackProcessStrategy.
